ebu_tt_live/bindings/test/test_role_mixin.py

```diff
@@ -0,0 +1,86 @@
+from unittest import TestCase
+
+from ebu_tt_live.bindings import d_div_type, d_br_type, d_body_type, d_p_type, d_span_type, div_type, \
+    br_type, body_type, p_type, span_type
+
+from ebu_tt_live.bindings._ebuttm import pMetadata_type, divMetadata_type
+
+
+class TestElementRoleMixin(TestCase):
+    class Metadata:
+        def __init__(self, role=None):
+            self.role = role
+
+    def setUp(self):
+        self.elements = {
+            'd_div': d_div_type(),
+            'd_br': d_br_type(),
+            'd_body': d_body_type(),
+            'd_p': d_p_type(),
+            'd_span': d_span_type(),
+            'div': div_type(),
+            'br': br_type(),
+            'body': body_type(),
+            'p': p_type(),
+            'span': span_type()
+        }
+        self.dataset = {'role_stack': []}
+        self.metadata_types = {
+            'div': divMetadata_type,
+            'p': pMetadata_type
```

Comment:
We should be able to test the metadata on more than just these two element types. Was there a problem doing that?

Reply:
yes, the other types bodyMetadata_type and spanMetadata_type are not callable. And there is no brMetadata_type

Reply:
That's very odd, you should be able to grab them:

```python
from ebu_tt_live.bindings._ebuttm import \
    divMetadata_type, bodyMetadata_type, pMetadata_type, spanMetadata_type, \
    metadataBase_type

bodyMetadata_element = bodyMetadata_type()
```

etc.

It looks like `d_br_type` should have a `metadata` element child that of type `d_metadata_type`.
Similarly `br_type` should have a `metadata` element child of type `anyMetadata_type`.


I'm seeing this by inspecting the PyXB-generated bindings code: searching in `ebu_tt_live/bindings/raw/__init__.py` for the text `._AddElement(pyxb.binding.basis.element(pyxb.namespace.ExpandedName(Namespace, 'metadata')` gives results like:

```python
body_type._AddElement(pyxb.binding.basis.element(pyxb.namespace.ExpandedName(Namespace, 'metadata'), _ImportedBinding_ebu_tt_live_bindings__ebuttm.bodyMetadata_type, scope=body_type, location=pyxb.utils.utility.Location('/Users/megitn02/Code/bbc/ebu-tt-live-toolkit/ebu_tt_live/xsd/ebutt_live.xsd', 155, 6)))
```

where the bit after the `.` in `_ImportedBinding_ebu_tt_live_bindings__ebuttm.bodyMetadata_type` is the class of the metadata object that relates to the element type to which the element is being added, in this example `bodyMetadata_type`.

Reply:
I could update for all types. I found my mistake why I was getting "not callable error". It was a typo issue. Now it is fixed.